timesketch/lib/analyzers/tagger.py

```diff
@@ -58,26 +66,62 @@ def tagger(self, name, config):
         """
         query = config.get('query_string')
         query_dsl = config.get('query_dsl')
-        create_view = config.get('create_view', False)
-        view_name = config.get('view_name', name)
+        save_search = config.get('save_search', False)
+        # For legacy reasons to support both save_search and
+        # create_view parameters.
+        if not save_search:
+            save_search = config.get('create_view', False)
+
+        search_name = config.get('search_name', None)
+        # For legacy reasons to support both search_name and view_name.
+        if search_name is None:
+            search_name = config.get('view_name', name)
+
         tags = config.get('tags', [])
         emoji_names = config.get('emojis', [])
         emojis_to_add = [emojis.get_emoji(x) for x in emoji_names]
 
+        expression_string = config.get('regular_expression', '')
+        expression = None
+        attributes = None
+        if expression_string:
+            try:
+                expression = re.compile(expression_string)
+            except re.error as exception:
+                # pylint: disable=logging-format-interpolation
+                logger.warning((
+                    'Regular expression [{0:s}] failed to compile, with '
+                    'error: {1!s}').format(expression_string, exception))
+                expression = None
+
+            attribute = config.get('re_attribute')
+            if attribute:
+                attributes = [attribute]
+
         event_counter = 0
-        events = self.event_stream(query_string=query, query_dsl=query_dsl)
+        events = self.event_stream(
+            query_string=query, query_dsl=query_dsl, return_fields=attributes)
 
         for event in events:
+            if expression:
+                value = event.source.get(attributes[0])
```

Comment:
Whats the idea with attributes[0] here - do you plan to support multiple attributes in the future?

Reply:
yes, that is the idea, to keep that open, since we might do that